packages/utils/src/dom/html.ts

```diff
@@ -211,17 +229,21 @@ export class HTMLDocument {
     }
 
     public get documentElement(): HTMLElement {
-        const htmlNode = this._document.children.find((node) => {
-            return node.type === 'tag' && node.name === 'html';
-        }) as ElementData;
-
-        return new HTMLElement(htmlNode, this);
+        return new HTMLElement(this._documentElement, this);
     }
 
     public get base(): string {
         return this._base;
     }
 
+    /**
+     * Check if this represents a template fragment as opposed to a full document.
+     */
+    public get isFragment(): boolean {
+        // Document is a fragment if `<html>` wasn't part of the original source.
+        return !this.originalDocument && !this._documentElement.sourceCodeLocation;
```

Comment:
Is it possible to not have `originalDocument` and not to have `_documentElement`?

Reply:
Yes, in fact most of the times we create the `HTMLDocument` without `originalDocument`. `originalDocument` is used in `jsdom` and `puppeteer` because we parsed twice:

1. original HTML down the wire
2. another time when we do the traversing to make sure we visit all possible new elements

It's the second time we send the `originalDocument` so we can find the position of the ones that were already present initially.

I also see why the name `originalDocument` could confuse anyone.

Reply:
Yep, maybe the name was confusing me.